[filebeat][streaming] - Added support for websocket keep_alive heartbeat, updated metrics & docs (#44204)
elastic/beats: docs/reference/filebeat/filebeat-input-streaming.md

Reviewer: Add metric docs for the change here.

  url: wss://localhost:443/_stream
```

> ## Keep Alive Configuration

Reviewer 2: ```suggestion
## Keep Alive configuration
```

Reviewer 2: We use sentence-style capitalization for headings.

> The `streaming` input currently supports keep-alive configuration options for streams of `type: websocket`. Using these configuration options we can further optimize the stability
> of our websocket connections and prevent them from idling out. 

Reviewer 2: ```suggestion
The `streaming` input currently supports keep-alive configuration options for streams of `type: websocket`. Use these configuration options to further optimize the stability
of your WebSocket connections and prevent them from idling out. 
```

The `keep_alive` setting has the following configuration options:

> * `enable`: This indicates whether keep-alive is enabled. By default, this is set to `false`.
> * `interval`: This is the interval between keep-alive messages. This is a time duration value and by default, this is set to `30s`.
> * `write_control_deadline`: This is the deadline for writing control frames (like `PING`, `PONG`, or `CLOSE`) on a websocket connection. This timeout helps prevent indefinite blocking when the server or client is not responding to control frame requests. This is a time duration value and by default, this is set to `10s`.

Reviewer 2: ```suggestion
* `enable`: Indicates whether keep-alive is enabled. By default, this is set to `false`.
* `interval`: Interval between keep-alive messages, expressed as a time duration value. The default value is `30s`.
* `write_control_deadline`: Deadline for writing control frames, like `PING`, `PONG`, or `CLOSE`, on a websocket connection. The timeout, expressed as a time duration value, helps prevent indefinite blocking when the server or client is not responding to control frame requests. The default value is `10s`.
```

Reviewer 2: `WebSocket` and `Keep-Alive` must be capitalized because they're proper nouns for their respective techs.

> ::::{note}
> It is recommended to not use `blanket_retries` and `infinite_retries` configuration options together with the `keep_alive` settings. The purpose of `keep_alive` is to keep the connection open so we do not need to `retry` and reconnect all the time. In some scenarios `keep_alive` might not work if the host websocket server is not configured to handle `ping` frames.
> ::::

Reviewer 2: ```suggestion
::::{note}
Don't use the `blanket_retries` and `infinite_retries` configuration options together with the `keep_alive` settings. The purpose of `keep_alive` is to keep the connection open so you don't need to `retry` and reconnect all the time. In some scenarios `keep_alive` might not work if the host WebSocket server is not configured to handle `ping` frames.
::::
```
